### ISSUE TITLE
Endless bugsplat appear

### ISSUE THREAD

**Issue by lochoang75** [mmtrt/leagueoflegends#39]
I use ubuntu 16.04 after install and select server. Game icon apear and bugsplat apper later. I try many time and it still not working.

**Comment by mmtrt** [comment]
can you post system info, drivers used, gpu OR is it laptop with intel + nvidia, OR is it dual display setup?
also post terminal log after launching game from leagueoflegends command.

**Comment by lochoang75** [comment]
I use Dell Inspiron 3543 with core i3 5005, intel HD graphics card , and dual monitor. 
This is my log 
[log.txt](https://github.com/mmtrt/leagueoflegends/files/3164553/log.txt)

**Comment by mmtrt** [comment]
try game with one monitor disabled.

**Comment by stuartfoxcroft** [comment]
I have this issue also, on a laptop with intel integrated graphics and a core i3, with a single display. I only discovered this issue after refreshing the leagueoflegends snap, and cannot even install the game, because the bugsplats appear instantly.

**Comment by mmtrt** [comment]
@stuartfoxcroft can you post info of snap info wine-platform | grep installed

**Comment by stuartfoxcroft** [comment]
stuart@STUART-W55XEU:~$ snap info wine-platform | grep installed
installed:   3.0.4-4.0.1-4.6            (142) 561MB -

**Comment by stuartfoxcroft** [comment]
I did initially get to play one game on 9.10 after running "snap refresh --candidate wine-platform", and then LoL stopped working after that, and refused to open. I then refreshed the snap in the hopes that this error would be fixed, and since then I have had constant bugsplats.

**Comment by mmtrt** [comment]
that looks fine can you post game log somewhere when its launched from terminal.

**Comment by stuartfoxcroft** [comment]
I have a pastebin of the console output from running LoL until the first bugsplat dialogue appeared: https://pastebin.com/wg4tSYVk

**Comment by mmtrt** [comment]
did your tried repairing the game.

**Comment by stuartfoxcroft** [comment]
I can't even open the client, and folders like ~/snap/leagueoflegends/common/ are completely empty.

**Comment by mmtrt** [comment]
there are hidden folders and files pressing ctril+h usually shows hidden files.

**Comment by mmtrt** [comment]
Is this game previously installed copy or just freshly install.

**Comment by stuartfoxcroft** [comment]
It's an attempt at a fresh install, but the installation process appears to be interrupted by the bugsplats.

**Comment by stuartfoxcroft** [comment]
![AfterReinstall](https://user-images.githubusercontent.com/50926932/58205027-830c2400-7cd5-11e9-929b-15b2c4e0002b.png)
![ScreenPriorToReinstall](https://user-images.githubusercontent.com/50926932/58205028-83a4ba80-7cd5-11e9-8fff-052ddbbea86b.png)

Both these images were taken with hidden files visible, the former being prior to reinstalling, and the latter being the current folder state.

**Comment by mmtrt** [comment]
Maybe try deleting `.wine` folder then try again and still that does not fix then to try to install the game with `wine-platform` snap from stable branch by updating it fixes download issue.
`snap refresh --stable wine-platform` .

**Comment by stuartfoxcroft** [comment]
I tried both methods, and both times I continued to receive bugsplats immediately.

**Comment by mmtrt** [comment]
Do you have wine install in your system without snap I mean if no then install missing pkgs from this guide https://www.gloriouseggroll.tv/how-to-get-out-of-wine-dependency-hell/

**Comment by stuartfoxcroft** [comment]
I had all of the recommended packages except for wine-staging and winehq-staging (the latter depending on the former).

**Comment by mmtrt** [comment]
hmm now try launching game from snap wineprefix with your system wine see if it installs.

**Comment by stuartfoxcroft** [comment]
How would I go about doing that?

**Comment by mmtrt** [comment]
`wine "$HOME/snap/leagueoflegends/common/.wine/drive_c/Riot Games/LeagueClient.exe"`

**Comment by stuartfoxcroft** [comment]
I get this error: "wine: cannot find '/home/stuart/snap/leagueoflegends/common/.wine/drive_c/Riot Games/LeagueClient.exe' "

**Comment by stuartfoxcroft** [comment]
Although if I navigate to that location, the file exists.

**Comment by stuartfoxcroft** [comment]
I have managed to run the client by manually navigating to the file and running it with wine.

**Comment by mmtrt** [comment]
yea you have to add inside of " " not with ' ' anyway good luck then if it fully installs then try launching it with snap.

**Comment by stuartfoxcroft** [comment]
It's fully installed now, gonna test it with snap now

**Comment by stuartfoxcroft** [comment]
The snap doesn't seem to work with --candidate wine-platform, it just bugsplats continually

**Comment by stuartfoxcroft** [comment]
But I can at least get into the client by running the LeagueClient.exe through system wine

**Comment by stuartfoxcroft** [comment]
Still unable to even launch the client through the snap, either on --candidate or --stable wine-platform.

**Comment by mmtrt** [comment]
maybe try using preinstalled game after reinstalling the snaps see if that solves anything though I'm not sure why many other users are affected by that bug could be side effects custom patched wine, glibc. If that still does not help then use lutris or system wine.

**Comment by stuartfoxcroft** [comment]
How would I reinstall the snaps without it deleting the snap data? And currently I can't seem to patch glibc for system wine, so I can't use system wine or lutris to actually get into a match.

**Comment by stuartfoxcroft** [comment]
At the moment my apparent options are: use system Wine, and be unable to enter a game because of the issues that started when 9.10 went live; use the snap launch, and get constant bugsplats and be unable to even load the client; not use anything, and waste the event pass that I paid for, and is non-refundable...

**Comment by mmtrt** [comment]
backup riot games folder from snap wineprefix then remove leagueoflegends snap with snap remove leagueoflegends and snap remove wine-platform then install both start league snap close region dialog move over riotgames folder to snap wineprefix and start game.

**Comment by stuartfoxcroft** [comment]
I've just tried that and experienced the same issues...

**Comment by mmtrt** [comment]
I'm compiling wine 4.6 if that does not fix that issue then I can't do anything fingers crossed.

**Comment by stuartfoxcroft** [comment]
I've tried removing and reinstalling the snaps multiple times, with wine-platform under --stable and --candidate, each time getting the same result i.e. bugsplats constantly

**Comment by mmtrt** [comment]
Compiled patched wine 4.6 and uploaded try wine-platform from --edge.
`snap refresh --edge wine-platform`

**Comment by Ads20000** [comment]
No luck for me even with `wine-platform` from `edge` and then reinstalling the League snap (revision 37 of League and 143 of Wine)

**Comment by stuartfoxcroft** [comment]
No luck here either

**Comment by stuartfoxcroft** [comment]
Update: Just managed to get into the client by changing my Windows version to Windows XP through leagueoflegends.winecfg

**Comment by Ads20000** [comment]
Nice! That works for me too! Will test.
Another possibility is using (connecting to a special snap providing?) Wine with [Wesie's 'alternative patch'](https://bugs.winehq.org/attachment.cgi?id=64507&action=diff) rather than using patched glibc. I've compiled that version of Wine (WoW64 build [via some advice on Reddit](https://www.reddit.com/r/leagueoflinux/comments/936rce/how_to_apply_the_new_patches_to_wine_staging/e3c0c1c/)) but haven't patched Lutris League to test it yet, gonna try the snap with Windows XP first.

**Comment by Ignacy-s** [comment]
Works on Ubuntu 16.04!
TL;DR: use `leagueoflegends.winecfg` to set system to windows XP (thanks stuart!)
Long:
Thank's so much Reza! You should give us a link to your paypal or use this new github feature that allows support for projects.

Works here on Ubuntu Mate 16.04 i5 2600k, nvidia 970, not sure if all steps were required but here's how it was achieved:
1. Read thoroughly subjects on reddit and this topic.
2. Updated my graphic drivers using nvidia PPA, from 396 to 418. (not sure if needed, but checking your graphics drivers doesn't hurt). (remember to do it from Ctrl-Alt-F1 terminal, and shut down your gui with sudo systemctl stop gnome or sth like that. after removing drivers relaunch it with systemctl start gnome/lightdm)
3. Purged all wine packages from system. Everything with wine went.
4. Reinstalled wine staging using https://wiki.winehq.org/Ubuntu
5. Removed both wine-platform and leagueoflegends snaps.
6. Installed leagueoflegends `snap install leagueoflegends --edge --devmode`
7. Refreshed wine-platform to stable to install the game `snap refresh --stable wine-platform`
9. Run leagueoflegends to install it. Eventually you can use system wine, as said by mmrt in this topic.
9. Refresh wine-platform into --edge build to enable new hacks `snap refresh --edge wine-platform`
10. Run `leagueoflegends.winecfg` and set the OS to Win XP
11. Run `leagueoflegends`. Works now.

**Comment by stuartfoxcroft** [comment]
I've made a new post in r/leagueoflinux to notify people of this fix. Glad to see that my trial and error (and a lot of guessing and stuff) has produced something functional lmao

**Comment by stuartfoxcroft** [comment]
Has anyone tried just running it with edge wine-platform and doing the winecfg step before running league?

**Comment by mmtrt** [comment]
Well thats good to hear at least it worked out but the question is how game works with setting os to XP though its support was ended by rito.

Anyway I've also created a wine patched appimage to sooth users getting the league working easily on linux.

https://github.com/mmtrt/Wine_Appimage/releases

**Comment by fabriciopirini** [comment]
> Works on Ubuntu 16.04!
> TL;DR: use `leagueoflegends.winecfg` to set system to windows XP (thanks stuart!)
> Long:
> Thank's so much Reza! You should give us a link to your paypal or use this new github feature that allows support for projects.
> 
> Works here on Ubuntu Mate 16.04 i5 2600k, nvidia 970, not sure if all steps were required but here's how it was achieved:
> 
> 1. Read thoroughly subjects on reddit and this topic.
> 2. Updated my graphic drivers using nvidia PPA, from 396 to 418. (not sure if needed, but checking your graphics drivers doesn't hurt). (remember to do it from Ctrl-Alt-F1 terminal, and shut down your gui with sudo systemctl stop gnome or sth like that. after removing drivers relaunch it with systemctl start gnome/lightdm)
> 3. Purged all wine packages from system. Everything with wine went.
> 4. Reinstalled wine staging using https://wiki.winehq.org/Ubuntu
> 5. Removed both wine-platform and leagueoflegends snaps.
> 6. Installed leagueoflegends `snap install leagueoflegends --edge --devmode`
> 7. Refreshed wine-platform to stable to install the game `snap refresh --stable wine-platform`
> 8. Run leagueoflegends to install it. Eventually you can use system wine, as said by mmrt in this topic.
> 9. Refresh wine-platform into --edge build to enable new hacks `snap refresh --edge wine-platform`
> 10. Run `leagueoflegends.winecfg` and set the OS to Win XP
> 11. Run `leagueoflegends`. Works now.

I did steps 6 to 11 here and worked!! Btw, I downgraded nvidia-driver from 430 to 415 (not sure if needed, prob not) just because it was the recommended on d9vk.

And I agree with the idea of Taqi Raza creating a PayPal or Patreon for donations ... you saved my league several times dude ... I would be happy to contribute!

**Comment by stuartfoxcroft** [comment]
@mmtrt wasn't there a fix for the anti-cheat stuff a while back that required XP?

**Comment by stuartfoxcroft** [comment]
maybe that's why it only works on XP

**Comment by mmtrt** [comment]
Yeah that could be, though rito is bringing new problems on each patch that is breaking hell loose on us lol.

**Comment by mmtrt** [comment]
Update wine-staging 4.9

pushed new update of `wine-platform` to edge branch

Unrelated to snap update:

Anyway I've also created a wine-staging league patched version appimage to sooth users getting the league working easily on linux.

https://github.com/mmtrt/Wine_Appimage/releases

**Comment by Ads20000** [comment]
Does `wine-platform` 3.0.4-4.0.1-4.6-4.9 (144) get this working on the Windows 7 setting? I found that revision 144 meant that League crashed after champion select and couldn't even get onto the game loading screen, had to revert to 3.0.4-4.0.1-4.6 (143).

**Comment by Ads20000** [comment]
Nevermind, this is documented at #45 I suppose I need to backup the wineprefix and then recreate it after upgrading back to revision 144.

**Comment by mmtrt** [comment]
Created simple patreon page. 
 https://www.patreon.com/mmtrt

**Comment by VasLem** [comment]
I cannot seem to make this work. The games installs fine, but once I try to initiate a custom game, it fails. The log is the following:

[log.txt](https://github.com/mmtrt/leagueoflegends/files/3266581/log.txt)

**Comment by VasLem** [comment]
Oh, I didnt see that there was an update today https://github.com/mmtrt/leagueoflegends/issues/7
After issuing the installation and refresh of the prescribed snaps, I managed to get into a custom game.
Thank you for all your efforts.

**Comment by mmtrt** [comment]
Yeah I was about to that here.
